src/uwtools/drivers/upp.py

```diff
@@ -81,6 +100,40 @@ def driver_name(cls) -> str:
         """
         return STR.upp
 
+    @property
+    def output(self) -> OutputT:
+        """
+        Returns a description of the file(s) created when this component runs.
+        """
+        # Derive values from the current driver config. GRIB output filename suffixes include the
+        # forecast leadtime, zero-padded to at least 2 digits (more if necessary). Avoid taking the
+        # log of zero.
+        cf = self.config["control_file"]
+        leadtime = int(self.leadtime.total_seconds() / 3600)
+        suffix = ".GrbF%0{}d".format(max(2, int(log10(leadtime or 1)) + 1)) % leadtime
+        # Read the control file into an array of lines. Get the number of blocks (one per output
+        # GRIB file) and the number of variables per block. For each block, construct a filename
+        # from the block's identifier and the suffix defined above.
+        try:
+            with open(cf, "r", encoding="utf-8") as f:
+                lines = f.read().split("\n")
+        except (FileNotFoundError, PermissionError) as e:
+            raise UWConfigError(f"Could not open UPP control file {cf}") from e
+        nblocks, lines = int(lines[0]), lines[1:]
```

Comment:
Wouldn't it be nice to `[h | rest]` this?

Reply:
Yes! I went looking for something suitable in the standard library and was momentarily excited about `isplit()` from `itertools`, but it doesn't do what I wanted. So I temporarily had
```
split = lambda x, i: (x[:i], x[i:])
```
but that returns two `list` objects, which I want in some places, but in others I want the scalar head value, and sometimes I want to apply a function to the result... so extra abstraction didn't seem worth it. But, someday: https://github.com/pytoolz/toolz